lib/aruba/environments/unix_environment.rb

```diff
@@ -0,0 +1,11 @@
+require 'aruba/environments/basic_environment'
+
+module Aruba
+  module Environments
+    class UnixEnvironment < BasicEnvironment
+      def match?(*)
+        Aruba::Platform.unix?
```

Comment:
Makes sense.  Where I think my point of view is different from yours is that it seems to me that you're mixing *implementation details* into modeling your domain ( the real world: the environment variables).  Yes, this gives you the results you need, but now you're no longer actually reflecting what an Environment really is.  Doing it this way, you're saying: We always have a UnixEnvironment, and in some cases we have *both* a UnixEnvironment and a WindowsEnvironment -- because both can be true (both can 'match'). That's not correct.  It's always one or the other, but not both (not according to what aruba is going to know and interact with, anyway).  
From my point of view, it's always best to model how things really are.  If we need to use implementation details/artifacts to get things accomplished, then that should be cleanly and obviously separated.  (There are lots of ways to accomplish that.  Ports and adapters is one, as it happens.)
This is a case where it's easy to model things as they really are, so it should be done.